### ISSUE TITLE
A--->B---C   cascading replication error

### ISSUE THREAD

**Issue by qq363960630** [2ndQuadrant/pglogical#88]
create database test;
create database test1;
create database test2;

\c test
create extension pglogical;
create table test (id int primary key, t text);
select * from pglogical.create_node('p1', 'host=127.0.0.1 port=55555 dbname=test');
select * from pglogical.replication_set_add_table('default', 'test');

\c test1
create extension pglogical;
create table test (id int primary key, t text);
select * from pglogical.create_node('s1', 'host=127.0.0.1 port=55555 dbname=test1');
select * from pglogical.replication_set_add_table('default', 'test');

select pglogical.create_subscription('sub1','host=127.0.0.1 port=55555 dbname=test','{default, default_insert_only}',false, true,'{all}');

\c test2
create extension pglogical;
create table test (id int primary key, t text);
select * from pglogical.create_node('s2', 'host=127.0.0.1 port=55555 dbname=test2');

select pglogical.create_subscription('sub2','host=127.0.0.1 port=55555 dbname=test1','{default, default_insert_only}',false, true,'{all}');

when  exec sql:  **insert into test values(1, '11');**

error log is:

LOG:  starting pglogical database manager for database test1
LOG:  manager worker [5527] at slot 2 generation 1 detaching cleanly
LOG:  starting apply for subscription sub1
LOG:  logical decoding found initial starting point at 0/16C0D10
DETAIL:  1 transaction needs to finish.
LOG:  logical decoding found consistent point at 0/172BB78
DETAIL:  Transaction ID 1560 finished; no more running transactions.
LOG:  exported logical decoding snapshot: "00000619-1" with 0 transaction IDs
LOG:  starting logical decoding for slot "pgl_test1_p1_sub1"
DETAIL:  streaming transactions committing after 0/172F9C0, reading WAL from 0/16C0C80
LOG:  logical decoding found initial starting point at 0/16C0D10
DETAIL:  1 transaction needs to finish.
LOG:  logical decoding found consistent point at 0/172BB78
DETAIL:  Transaction ID 1560 finished; no more running transactions.
LOG:  manager worker [5538] at slot 3 generation 1 detaching cleanly
LOG:  manager worker [5539] at slot 4 generation 1 detaching cleanly
LOG:  starting pglogical database manager for database test2
LOG:  starting apply for subscription sub2
LOG:  logical decoding found consistent point at 0/1735248
DETAIL:  There are no running transactions.
LOG:  exported logical decoding snapshot: "00000622-1" with 0 transaction IDs
LOG:  starting logical decoding for slot "pgl_test2_s1_sub2"
DETAIL:  streaming transactions committing after 0/1735280, reading WAL from 0/1735248
LOG:  logical decoding found consistent point at 0/1735248
DETAIL:  There are no running transactions.
ERROR:  replication origin with OID 1 is already active for PID 5531
LOG:  apply worker [5544] at slot 3 generation 2 crashed
LOG:  starting apply for subscription sub2
LOG:  terminating walsender process due to replication timeout
LOG:  terminating walsender process due to replication timeout
LOG:  manager worker [5628] at slot 4 generation 2 detaching cleanly
LOG:  manager worker [5629] at slot 4 generation 3 detaching cleanly

now  database test1 is in the dead cycle

`

**Comment by PJMODOS** [comment]
Yes, cascading is not supported within single postgresql instance.

**Comment by qq363960630** [comment]
thanks
